### DISCUSSION TITLE
Does LangChain support OpenAI's Batch API?

### DISCUSSION THREAD

**Discussion by ScoutLead** [langchain4j/langchain4j#2180] [Q&A]
![image](https://github.com/user-attachments/assets/09f50704-3bb7-4981-8ed2-cac2b972542d)

**Comment by mike-adonis** [comment]
@ScoutLead, did you get any answers?

**Comment by zharkomi** [comment]
I'm using batch api of Open AI. Will it be helpful to integrate it into this library?

**Comment by dliubarskyi** [comment]
cc @Bram--

**Comment by Bram--** [comment]
Currently OpenAI's batch API is not supported. There is support for the Gemini Batch API([Chat](https://docs.langchain4j.dev/integrations/language-models/google-ai-gemini#batch-processing), [Embedding](https://docs.langchain4j.dev/integrations/embedding-models/google-ai-gemini#batch-embedding-processing) & [Image](https://docs.langchain4j.dev/integrations/image-models/gemini#batch-image-generation)) and I'm in the [process of generalizing this](https://github.com/langchain4j/langchain4j/pull/4529) and implementing it for [Anthropic's Claude](https://github.com/langchain4j/langchain4j/pull/4394).

Once #4529 is in it'd be great for someone to pick up the integration for OpenAI's batch API as I personally don't use their models. Issue: #3916

**Reply by zharkomi** [reply]
Interfaces look good. When are you going to merge it?

**Reply by Bram--** [reply]
Just awaiting approvals
